src/plugins/data_view_editor/public/components/data_view_editor_flyout_content.tsx

```diff
@@ -157,11 +189,13 @@ const IndexPatternEditorFlyoutContentComponent = ({
     const getTitles = async () => {
       const indexPatternTitles = await dataViews.getTitles();
 
-      setExistingIndexPatterns(indexPatternTitles);
+      setExistingIndexPatterns(
+        editData ? indexPatternTitles.filter((v) => v !== editData.title) : indexPatternTitles
```

Comment:
I think there is a bug with `indexPatternTitles` cache. 
1. edit a data view title 
2. Save
3. Edit it again using the older value
4. It says that such data view exists

Reply:
fixed.